src/api/processors/saturate.cpp

```diff
@@ -0,0 +1,43 @@
+#include "processors/saturate.h"
+
+namespace weserv {
+namespace api {
+namespace processors {
+
+VImage Saturate::process(const VImage &image) const {
+    auto mult = query_->get_if<float>("sat", [](float m) {
+                                                // multiplier needs to be
+                                                // in range of 0 - 10000
+                                                return m >= 0 && m <= 10000;
+                                            },
+                                            1.0F);
+
+    // Should we process the image?
+    if (mult == 1.0F) {
+        return image;
+    }
+
+    // Define saturation matrix
+    // clang-format off
+    std::array<double, 9> saturate = {
+        0.213f + 0.787f * mult, 0.715f - 0.715f * mult, 0.072f - 0.072f * mult,
+        0.213f - 0.213f * mult, 0.715f + 0.285f * mult, 0.072f - 0.072f * mult,
+        0.213f - 0.213f * mult, 0.715f - 0.715f * mult, 0.072f + 0.928f * mult
+    };
+
+    return image
+                .colourspace(VIPS_INTERPRETATION_sRGB)
+                .recomb(image.bands() == 3
+                        ? VImage::new_from_memory(saturate.begin(), 9 * sizeof(double), 3,
+                                                  3, 1, VIPS_FORMAT_DOUBLE)
+                        : VImage::new_matrixv(4, 4,
+                                              saturate[0], saturate[1], saturate[2], 0.0,
+                                              saturate[3], saturate[4], saturate[5], 0.0,
+                                              saturate[6], saturate[7], saturate[8], 0.0,
+                                              0.0, 0.0, 0.0, 1.0));
+    // clang-format on
```

Comment:
How about exposing libvips' recombination matrix operation instead? Something like this:
```suggestion
    auto recomb = query_->get<std::vector<float>>("recomb", {});

    // Should we process the image?
    if (recomb.empty()) {
        return image;
    }

    // Need to convert std::vector<float> to std::vector<double>
    std::vector<double> m(recomb.begin(), recomb.end());

    // Recomb the image with the specified matrix.
    // clang-format off
    auto matrix = image.bands() == 3
                      ? VImage::new_from_memory(m.data(), 9 * sizeof(double), 3,
                                                3, 1, VIPS_FORMAT_DOUBLE)
                      : VImage::new_matrixv(4, 4,
                                            m[0], m[1], m[2], 0.0,
                                            m[3], m[4], m[5], 0.0,
                                            m[6], m[7], m[8], 0.0,
                                            0.0, 0.0, 0.0, 1.0);

    return image
        .colourspace(VIPS_INTERPRETATION_sRGB)
        .recomb(matrix);
    // clang-format on
```

So that users can fill in a 3x3 recombination matrix of their choice. For example, a sepia filter could look like this:
```
https://images.weserv.nl/static/zebra.jpg?recomb=0.3588,0.7044,0.1368,0.2990,0.5870,0.1140,0.2392,0.4696,0.0912
```

This is effectively the same as `<feColorMatrix>` in SVG.